### DISCUSSION TITLE
How to filter between dates in prisma orm without using raw query?

### DISCUSSION THREAD

**Discussion by lykemike** [prisma/prisma#11443] [Q&A] [answered]
```ts
import { PrismaClient } from ".prisma/client";
const prisma = new PrismaClient();

// THIS WORKS, IT RETURNS 33 ARRAYS
export const getJournal = async (start_date, end_date) => {
  return await prisma.journalTransactions.findMany({
    where: {
      date: {
        lte: 2022-01-01,
        gte: 2022-01-30,
      },
    },
  });
};

// THIS DOESNT WORK, IT RETURNS 0 ARRAYS
export const getJournal = async (start_date, end_date) => {
  return await prisma.journalTransactions.findMany({
    where: {
      date: {
        lte: 2022-01-15,
        gte: 2022-01-30,
      },
    },
  });
};
```

// how do i achieve SELECT * FROM "JournalTransactions" WHERE date BETWEEN '2022-01-15' AND '2022-01-30' ORDER BY id DESC; this in // prisma orm

**Comment by austincrim** [comment, marked as answer]
Hey @lykemike 👋 ,

If you are looking for a range between 2022-01-15 and 2022-01-30, you would need to reverse your condition:

```ts
return await prisma.journalTransactions.findMany({
    where: {
      date: {
        lte: new Date("2022-01-30"),
        get: new Date("2022-01-15"),
      },
    },
  });
```

Give that a shot and let me know if it works!

**Reply by lykemike** [reply]
Thank you, it works!!

**Reply by azabraao** [reply]
I don't understand how did this work for you guys as it has plain numbers in the dates. Shouldn't it be a string or something?

Like: 
```javascript
date: {
    lte: "2022-01-30",
    gte: "2022-01-15",
},
```
Let me know where I got it wrong.

**Reply by wolfganggreschus** [reply]
Hey azabro,
you can convert the strings like this:
`date: {
    lte: new Date("2022-01-30"),
    gte: new Date("2022-01-15"),
},`

**Reply by Ref-Bit** [reply]
Just to expand on @wolfganggreschus answer, if want to produce a Date ISO string, you should chain the ```toISOString``` method for a valid DateTime stamp like this:
```javascript
date: {
  lte: new Date("2022-01-30").toISOString(), // "2022-01-30T00:00:00.000Z"
  gte: new Date("2022-01-15").toISOString(), // "2022-01-15T00:00:00.000Z"
}
```

**Reply by RashJrEdmund** [reply]
thank you sir, the Date's .toISOString() method was lacking in mine, and made my query not work, this solved it.

**Comment by lucianobs1** [comment]
what is lte ,gt  and gte ? I looked in the documentation but couldn't find what it is. And the difference between them

**Reply by lykemike** [reply]
**LTE: Less Than or Equal to**
_e.g filtering/sorting a post Less Than or Equal to 10 likes ( <= 10)_
Symbol of Less Than or Equal to <=

**GT: Greater Than**
_e.g filtering/sorting a post Greater Than to 10 likes ( > 10 )_
Symbol of Greater Than >

**GTE: Greater Than or Equal to**
_e.g filtering/sorting a post Greater Than or Equal to 10 likes ( >= 10 )_
Symbol of Greater Than or Equal to  >=

LTE <= takes all numbers starting from 10 and below [10, 9, 8, etc...]
GT > takes all numbers above 10, the number 10 is not included [11, 12, 13, etc...]
GTE >= takes all numbers starting from 10 and above [10, 11, 12, etc...]

https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting

**Reply by ThaJay** [reply]
@lucianobs1 Yes I agree that is not obvious.

Abbreviations in code are a terrible pattern.

**Reply by ffakira** [reply]
> @lucianobs1 Yes I agree that is not obvious.
> 
> Abbreviations in code are a terrible pattern.

Actually `gt`, `lt`, `gte` and `lte` abbreviations are quite common in ORM libraries.

**Reply by ThaJay** [reply]
@ffakira
"For every person that complains, there will be 10 that stayed silent."
"Explicit is better than implicit."
"Readability is the first purpose of code."

**Reply by james-langridge** [reply]
Here's a link to the `gte` API reference with a relevant example `Get all Post records where date_created is greater than March 19th, 2020`: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#get-all-post-records-where-date_created-is-greater-than-march-19th-2020

**Comment by elena-calcada** [comment]
Hey guys! This discussion helped me more than you can imagine.

I want to share the link of the prism documentation where such abbreviations (lt, lte, gt and gte) are presented.

This documentation is very interesting. For me, it is. It is an excellent reference that will make filters easier.

Follow the link:

https://www.prisma.io/docs/reference/api-reference/prisma-client-reference

**Comment by Hasankanso** [comment]
but...how to compare with other columns values?

**Comment by Hasankanso** [comment]
After digging I found this test

https://github.com/prisma/prisma/commit/83100ad037a223d8ad747bf348308890831c956d#diff-4e1a897b4d46d7a0ff25c0535aa456af8fa0d34728e0b54d99e2d8dad0447246

```
const products = prisma.product.findMany({
          where: {
            properties1: {
              equals: prisma.product.fields.title  // prisma.<table-name>.fields.<column-name>
               }
          }
       });
```

**Reply by sorenbs** [reply]
Here is the documentation: https://www.prisma.io/docs/orm/reference/prisma-client-reference#compare-columns-in-the-same-table

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
